mypy/maptype.py

```diff
@@ -1,8 +1,19 @@
 from typing import Dict, List
 
+import mypy.typeops
```

Comment:
This causes a circular import:
```
  File "venv/lib/python3.10/site-packages/mypy/report.py", line 20, in <module>
    from mypy import stats
  File "venv/lib/python3.10/site-packages/mypy/stats.py", line 13, in <module>
    from mypy.argmap import map_formals_to_actuals
  File "venv/lib/python3.10/site-packages/mypy/argmap.py", line 8, in <module>
    from mypy.maptype import map_instance_to_supertype
  File "venv/lib/python3.10/site-packages/mypy/maptype.py", line 3, in <module>
    import mypy.typeops
  File "venv/lib/python3.10/site-packages/mypy/typeops.py", line 15, in <module>
    from mypy.maptype import map_instance_to_supertype
ImportError: cannot import name 'map_instance_to_supertype' from partially initialized module 'mypy.maptype' (most likely due to a circular import) (venv/lib/python3.10/site-packages/mypy/maptype.py)
```

Reply:
Thanks for reporting! Do you have a (preferably simple) repro for this?

Reply:
Simply importing `mypy.report` without mypyc enabled should trigger this issue.

Reply:
See https://github.com/python/mypy/pull/13809 Btw just curious, why do you need to `import mypy.report` directly?

Reply:
It's one of [sanity checks in nixpkgs](https://github.com/NixOS/nixpkgs/blob/0490b307e5556a8804e710d0c744d29c80fbce48/pkgs/development/python-modules/mypy/default.nix#L66), to make sure mypy module is not broken.